src/tools/rust-analyzer/crates/base-db/src/lib.rs

```diff
@@ -1,4 +1,11 @@
 //! base_db defines basic database traits. The concrete DB is defined by ide.
+
+#![cfg_attr(feature = "in-rust-tree", feature(rustc_private))]
+
+#[cfg_attr(all(feature = "in-rust-tree", test), allow(unused_extern_crates))]
+#[cfg(all(feature = "in-rust-tree", test))]
+extern crate rustc_driver;
```

Comment:
This beautiful hack was needed for all participating r-a crates under test scenario

Reply:
😢 our poor `lib.rs` files

We should at least be able to do 
```diff
-#[cfg_attr(all(feature = "in-rust-tree", test), allow(unused_extern_crates))]
#[cfg(all(feature = "in-rust-tree", test))]
-extern crate rustc_driver;
+extern crate rustc_driver as _;
```

Reply:
Fixed to use that